src/evaluators/ClassDeclaration.js

```diff
@@ -10,11 +10,272 @@
 /* @flow */
 
 import type { Realm } from "../realm.js";
-import type { LexicalEnvironment } from "../environment.js";
+import type { LexicalEnvironment, Reference } from "../environment.js";
 import type { Value } from "../values/index.js";
-import type { Reference } from "../environment.js";
-import type { BabelNodeClassDeclaration } from "babel-types";
+import { NullValue, EmptyValue, ObjectValue } from '../values/index.js';
+import type { BabelNodeClassDeclaration, BabelNodeExpression, BabelNodeClassMethod } from "babel-types";
+import parse from "../utils/parse.js";
+import {
+  HasOwnProperty,
+  SetFunctionName,
+  IsConstructor,
+  Get,
+  MakeConstructor,
+  CreateMethodProperty,
+  DefineMethod,
+  NewDeclarativeEnvironment,
+  MakeClassConstructor,
+  ObjectCreate,
+  ConstructorMethod,
+  GetValue,
+  IsStatic,
+  PropertyDefinitionEvaluation,
+  InitializeBoundName,
+  NonConstructorMethodDefinitions,
+} from "../methods/index.js";
+import { AbruptCompletion } from "../completions.js";
+import invariant from "../invariant.js";
 
+function EvaluateClassHeritage(realm: Realm, ClassHeritage: BabelNodeExpression, strictCode: boolean): ObjectValue | null {
+  let ref = realm.getRunningContext().lexicalEnvironment.evaluate(ClassHeritage, strictCode);
+  let val = GetValue(realm, ref);
+  if (!(val instanceof ObjectValue)) {
```

Comment:
In additional to regular "concrete" values, Prepack also allows for "abstract" values, which are implemented via additional subtypes (`AbstractValue` also derives from `Value`).

To get us back into a safe zone where we can reason exactly about values, you should always call `.throwIfNotConcrete` on a value before checking its subtype via `instanceof`. The same comment goes for a few more `instanceof` checks I've seen below.

(Later, once the concrete behavior is all working, we can look into actually supporting reasoning about abstract values.)